KeyboardHelperTests/Tests/KeyboardHelperTests.swift

```diff
@@ -9,6 +9,93 @@
 import XCTest
 @testable import KeyboardHelper
 
+class ShowSpyDelegate : KeyboardNotificationDelegate {
+    var kai : KeyboardAppearanceInfo?
+    
+    var expectation : XCTestExpectation?
+    
+    func keyboardWillDisappear(info: KeyboardAppearanceInfo) {
+        
+    }
+    
+    func keyboardWillAppear(info: KeyboardAppearanceInfo) {
+        guard let expectation = expectation else {
+            XCTFail("ShowSpyDelegate was not setup correctly. Missing XCTExpectation reference")
+            return
+        }
+        
+        kai = info
+        expectation.fulfill()
+    }
+}
+
+class HideSpyDelegate : KeyboardNotificationDelegate {
+    var kai : KeyboardAppearanceInfo?
+    
+    var expectation : XCTestExpectation?
+    
+    func keyboardWillDisappear(info: KeyboardAppearanceInfo) {
+        guard let expectation = expectation else {
+            XCTFail("HideSpyDelegate was not setup correctly. Missing XCTExpectation reference")
+            return
+        }
+        
+        kai = info
+        expectation.fulfill()
+    }
+    
+    func keyboardWillAppear(info: KeyboardAppearanceInfo) {
+        
+    }
+}
+
 class KeyboardHelperTests: XCTestCase {
 
+    func testKeyboardShowDelegateMethodIsCalledAsync() {
+        let spyDelegate = ShowSpyDelegate()
+        let kh = KeyboardHelper(delegate: spyDelegate)
+        
+        let expectation = expectationWithDescription("KeyboardHelper calls the delegate as the result of receiving the show notification")
+        spyDelegate.expectation = expectation
+
+        NSNotificationCenter.defaultCenter().postNotificationName(UIKeyboardWillShowNotification, object: kh)
```

Comment:
It would be nice to send along custom `KeyboardApperanceInfo` and test it in the specific class (ShowSpy, HideSpy) for the same values, just so they don't get somehow changed by `NSNotificationCenter` or something :0